File: GrayMattr.Samples.Csharp/YamlStringify.cs
namespace GrayMattr.Samples.Csharp.YamlStringify;

public class YamlStringify
{
    static void Main()
    {
        // Stringify back to YAML

        GrayFile<IDictionary<string, string>> file1 = Mattr.Parse<IDictionary<string, string>>(
            string.Join("\n", ["---", "foo: bar", "---", "This is content"])
        );

        file1.Stringify();

        // custom data

        GrayFile<IDictionary<string, string[]>> file2 = Mattr.Parse<IDictionary<string, string[]>>(
            string.Join("\n", ["This is content"])
        );

        file2.Stringify(new Dictionary<string, string[]>() { { "baz", ["one", "two", "three"] } });
    }
}

File: GrayMattr.Samples/YamlStringify.fs
module YamlStringify

open GrayMattr

// Stringify back to YAML

let file1: GrayFile<obj> =
    Mattr.Parse([| "---"; "foo: bar"; "---"; "This is content" |] |> String.concat "\n")

file1.Stringify() |> ignore

// custom data

let file2: GrayFile<obj> =
    Mattr.Parse([| "This is content" |] |> String.concat "\n")

(file2.Stringify(dict [ "baz", [| "one"; "two"; "three" |] ])) |> ignore
